astropy/nddata/mixins/ndarithmetic.py

```diff
@@ -556,59 +514,97 @@ def _arithmetic_meta(self, operation, operand, handle_meta, **kwds):
-        result, kwargs = self._arithmetic(np.subtract, operand, **kwargs)
-        return self.__class__(result, **kwargs)
+        return cls(result, **kwargs)
 
+    @sharedmethod
     @format_doc(_arit_doc, name="multiplication", op="*")
-    def multiply(self, operand, **kwargs):
-        result, kwargs = self._arithmetic(np.multiply, operand, **kwargs)
-        return self.__class__(result, **kwargs)
+    def multiply(self, operand, operand2=None, **kwargs):
 
+        operand, operand2, kwargs, cls = self._resolve(operand, operand2, **kwargs)
+
+        result, kwargs = operand._arithmetic(np.multiply, operand2, **kwargs)
+
+        return cls(result, **kwargs)
+
+    @sharedmethod
     @format_doc(_arit_doc, name="division", op="/")
-    def divide(self, operand, **kwargs):
-        result, kwargs = self._arithmetic(np.true_divide, operand, **kwargs)
-        return self.__class__(result, **kwargs)
-
-    # TODO: Sharedmethod instead of classmethod? (embray)
-    # http://docs.astropy.org/en/v1.0.5/api/astropy.utils.decorators.sharedmethod.html#astropy.utils.decorators.sharedmethod
-    # But that only allows for one documentation and we need the first operand.
-    # Also it doesn't allow sphinx to build two documentations one for the
-    # classmethod and one of the instance method and since creating one
-    # doc that applies to both is not-trivial I set this option back for now.
-    # So I decided to temporarly alter the name that there are not too obvious
-    # name clashes by adding a 'ic_' prefix that stands for 'interclass'
-    # arithmetic operations
-
-    @classmethod
-    @format_doc(_arit_cls_doc, 'add')
-    def ic_addition(cls, operand1, operand2, **kwargs):
-        # Convert the first operand to the implicit passed class (cls)
-        # this allows for reverse operations.
-        operand1 = cls(operand1)
-        # Call the instance function for addition to proceed
-        return operand1.add(operand2, **kwargs)
-
-    @classmethod
-    @format_doc(_arit_cls_doc, 'subtract')
-    def ic_subtraction(cls, operand1, operand2, **kwargs):
-        operand1 = cls(operand1)
-        return operand1.subtract(operand2, **kwargs)
-
-    @classmethod
-    @format_doc(_arit_cls_doc, 'multiply')
-    def ic_multiplication(cls, operand1, operand2, **kwargs):
-        operand1 = cls(operand1)
-        return operand1.multiply(operand2, **kwargs)
-
-    @classmethod
-    @format_doc(_arit_cls_doc, 'divide')
-    def ic_division(cls, operand1, operand2, **kwargs):
-        operand1 = cls(operand1)
-        return operand1.divide(operand2, **kwargs)
+    def divide(self, operand, operand2=None, **kwargs):
+
+        operand, operand2, kwargs, cls = self._resolve(operand, operand2, **kwargs)
+
+        result, kwargs = operand._arithmetic(np.true_divide, operand2, **kwargs)
+
+        return cls(result, **kwargs)
+
+    @sharedmethod
+    def _resolve(self_or_cls, operand, operand2, **kwargs):
```

Comment:
I'm not very happy with this method-name but in theory it isn't important when subclassing. But I'd appreciate any suggestions.